src/language-js/printer-estree.js

```diff
@@ -4594,8 +4594,13 @@ function printTypeParameters(path, options, print, paramsKey) {
           grandparent.type === "TSTypeAnnotation" &&
           n[paramsKey][0].type !== "TSUnionType" &&
           n[paramsKey][0].type !== "UnionTypeAnnotation" &&
+          n[paramsKey][0].type !== "TSIntersectionType" &&
```

Comment:
These conditions currently catch two different situations:

1. Variable declarations with generic type annotations: `const a: Type<Params> = value;` like the second and third code samples in #6899.
2. Variable declarations initialized with an arrow function, for which a generic return type is specified: `const a = (params): Type<Params> => {}` like #6886 and the first code sample in #6899.

I think the right thing to do here is to exclude the situation # 2, completely or at least when the arrow function has parameters (because the parameters can break the line). I think when you made #6467, you didn't keep the situation # 2 in mind at all, did you?

Reply:
You're right. Sorry, I didn't keep the situation # 2 in mind when I created #6467. I pushed commits to exclude the situation # 2.